src/main/java/formflow/library/ScreenController.java

```diff
@@ -339,10 +347,19 @@ private ModelAndView handlePost(
           )
       );
       submission.setSubmittedAt(OffsetDateTime.now());
+
+      if (shortCodeConfig.isCreateShortCodeAtSubmission()) {
+        submissionRepositoryService.generateAndSetUniqueShortCode(submission);
+      }
     }
 
     actionManager.handleBeforeSaveAction(currentScreen, submission);
     submission = saveToRepository(submission);
+
```

Comment:
I think this needs to be in the POST endpoint and not the POST + /submit endpoint.

As is right now, the at creation method would not work correctly since this endpoint is only hit when submitting the application. You want the POST mapping which starts on line 234.

Reply:
Also, though unlikely, it's _possible_ at some point someone will create an application where the flow starts with a subflow and the first POST is in a subflow, which makes me wonder if we should account for that by adding the on creation but to the POST subflow endpoint as well (starts on 448) ?

Reply:
Ok, these should be updated per our Tuple!